### ISSUE TITLE
Check for unused keyword arguments

### ISSUE THREAD

**Issue by jonniedie** [tonyhffong/Lint.jl#270]
Only positional arguments are currently checked. It might also be worth adding an extra stern warning for unused keyword argument slurps, because they can cause silent failures on misspelled keyword arguments:
```julia
julia> badfun(;some_arg=1, another_arg=2, kwargs...) = some_arg*another_arg^2
badfun (generic function with 1 method)

julia> badfun(;another_arg=5)
25

julia> badfun(;anotherarg=5)
4

julia> betterfun(;some_arg=1, another_arg=2) = some_arg*another_arg^2
betterfun (generic function with 1 method)

julia> betterfun(;another_arg=5)
25

julia> betterfun(;anotherarg=5)
ERROR: MethodError: no method matching betterfun(; anotherarg=5)
Closest candidates are:
  betterfun(; some_arg, another_arg) at REPL[2]:1 got unsupported keyword argument "anotherarg"
Stacktrace: ...
```
